File: b860(weedle)/Python/weedle.py
from sys import stdin, stdout
while True:
    line = stdin.readline()
    if not line: break
    candies, weedles = map(int, line.split())
    total_evolution = 0
    while weedles > 0:
        evolution = min(candies // 12, weedles)
        total_evolution += evolution
        candies -= evolution * 10
        weedles -= evolution
        if candies < 12:
            weedles -= 12 - candies
            candies = 12
    stdout.write(f'{total_evolution}\n')

File: b860(weedle)/C++/weedle.cpp
#include <iostream>

#define min(x, y) x < y ? x : y

int main(int argc, char const *argv[]) {
  int candies, weedles, evolution, total_evolution;
  while (std::cin >> candies >> weedles) {
    total_evolution = 0;
    while (weedles > 0) {
      evolution = min(candies / 12, weedles);
      candies -= evolution * 10;
      weedles -= evolution;
      total_evolution += evolution;
      if (candies < 12) {
        weedles -= 12 - candies;
        candies = 12;
      } 
    }
    std::cout << total_evolution << '\n';
  }
  return 0;
}
